parcels/uxgrid.py

```diff
@@ -15,34 +15,56 @@ class UxGrid(BaseGrid):
     for interpolation on unstructured grids.
     """
 
-    def __init__(self, grid: ux.grid.Grid) -> UxGrid:
+    def __init__(self, grid: ux.grid.Grid, z: np.ndarray) -> UxGrid:
+        """
+        Initializes the UxGrid with a uxarray grid and vertical coordinate array.
+
+        Parameters
+        ----------
+        grid : ux.grid.Grid
+            The uxarray grid object containing the unstructured grid data.
+        z : np.ndarray
+            A 1D array of vertical coordinates (depths) corresponding to the vertical position of layer faces of the grid
```

Comment:
Perhaps this should be updated to an `xr.DataArray` object so that labels are maintained (motivated by https://github.com/OceanParcels/Parcels/issues/2031#issuecomment-2958323090)

Reply:
Also, depending on the result of the comment we should update this `__init__`

e.g., "Yes, uxarray allows nz to be spatially and temporally varying. For now though, let's only consider the case where its non spatially/temporally varying, and leave that open for future development."

In that case it would be good to update the `init` to ensure the time and face ID dims aren't in `nz` so that if people try to provide spatially varying nz they get a `NotImplementedError`.

Reply:
Yea, I think a `DataArray` would be better. I think it'd be ok to initially support only fixed in space and time vertical coordinate. This will work for flat bottomed topography and should be sufficient for demonstration. After getting through baroclinic gyre examples, it'd be good to work with the FESOM2 folks to add topography to that simulation to test out `nz` as a function of `faceID` (and get it implemented of course).

I'll get this done this morning